data/settings/src/main/java/com/google/jetpackcamera/settings/LocalSettingsRepository.kt

```diff
@@ -105,4 +116,17 @@ class LocalSettingsRepository @Inject constructor(
                 .build()
         }
     }
+
+    override suspend fun updateAspectRatio(aspectRatio: AspectRatio) {
+        val newStatus = when (aspectRatio) {
+            AspectRatio.NINE_SIXTEEN -> AspectRatioProto.ASPECT_RATIO_NINE_SIXTEEN
+            AspectRatio.THREE_FOUR -> AspectRatioProto.ASPECT_RATIO_THREE_FOUR
+            AspectRatio.ONE_ONE -> AspectRatioProto.ASPECT_RATIO_ONE_ONE
+        }
+        jcaSettings.updateData { currentSettings ->
+            currentSettings.toBuilder()
+                .setAspectRatioStatus(newStatus)
+                .build()
+        }
```

Comment:
We can use the more kotlin idiomatic version with the `copy` function.

```
currentSettings.copy {
     aspectRatio = newAspectRatio
}
```

Reply:
i had written the update functions like that previously, but it seems the best practice way is to follow the builder method